### PR TITLE
Simplify org.eclipse.equinox.common.tests

### PR DESCRIPTION
**Pull request by HannesWell**
Simply run all tests in the plug-in instead of specifying suits that
contain all tests. This also simplifies the addition of new tests.

### CODE REVIEW

File: bundles/org.eclipse.equinox.common.tests/src/org/eclipse/equinox/common/tests/AllTests.java [hunk 1]
```diff
@@ -1,36 +0,0 @@
-/*******************************************************************************
```
Comment (akurtakov) [inline bundles/org.eclipse.equinox.common.tests/src/org/eclipse/equinox/common/tests/AllTests.java:1]: If you remove this class tests can no longer be run in I-builds as it's waht is set in test.xml https://github.com/eclipse-equinox/equinox/blob/master/bundles/org.eclipse.equinox.common.tests/test.xml#L48

**Review by akurtakov — CHANGES_REQUESTED** [review]
Tests should be runnable without tycho/maven and this is what I-builds does to run on all supported OS/arch combos. 
Either test framework should be extended to work without having suite or this class should stay.

File: bundles/org.eclipse.equinox.common.tests/src/org/eclipse/equinox/common/tests/AllTests.java (hunk 1 above)
Comment (HannesWell) [reply bundles/org.eclipse.equinox.common.tests/src/org/eclipse/equinox/common/tests/AllTests.java:1]: I'm a bit surprised. I assumed that the test.xml are just leftovers from the past where Maven was not used and wanted to suggest to remove them.
How are these test.xml files picked up in an I-build? AFAIK Tycho-surefire does not respect them and either runs the entire plug-in or the suite specified in corresponding properties.

Comment (akurtakov) [reply bundles/org.eclipse.equinox.common.tests/src/org/eclipse/equinox/common/tests/AllTests.java:1]: https://wiki.eclipse.org/Platform-releng/Eclipse_Test_Framework (although probably outdated in places) should explain things. https://github.com/eclipse-platform/eclipse.platform.releng/blob/master/bundles/org.eclipse.test/src/org/eclipse/test/EclipseTestRunner.java is the key part IIRC.

**Comment by HannesWell** [comment]
> Tests should be runnable without tycho/maven and this is what I-builds does to run on all supported OS/arch combos. Either test framework should be extended to work without having suite or this class should stay.

I have to admit I never noticed that most (but not all?) test plug-ins have a test.xml, I noticed them occasionally but assumed they are just ancient leftovers. Can you explain why they are necessary and why Tycho cannot simply run them just with the default tycho:surefire that just executes all tests?

File: bundles/org.eclipse.equinox.common.tests/src/org/eclipse/equinox/common/tests/AllTests.java (hunk 1 above)
Comment (akurtakov) [reply bundles/org.eclipse.equinox.common.tests/src/org/eclipse/equinox/common/tests/AllTests.java:1]: Maven is not used for running the tests for I-builds as they are run on already provisioned target and this is done on 6 different os/arch/jvm combos without duplicating the whole rebuild process on each combo as maven would do.

**Comment by mickaelistria** [comment]
"Official" tests, the ones that sends the emails, are run with antrunner against the whole SDK. Related jobs are available at https://ci.eclipse.org/releng/view/Automated%20tests/ . Tycho may be able to do the job, but that would require some more tweaks which so far never managed to become a priority

**Comment by HannesWell** [comment]
Thank you both for the explanation. I wasn't aware of that.
But looking at it naively I suspect that it should be possible to run the entire plug-in by using `selectDirectory()` and passing the test plug-ins root path (is it a directory or a jar?) if no `testClassName` is specified here:
https://github.com/eclipse-platform/eclipse.platform.releng/blob/7768dbe43bbb01fbbb095ed0b79911522af819c7/bundles/org.eclipse.test/src/org/eclipse/test/EclipseTestRunner.java#L214

I will have a more detailed look at it tomorrow.

**Comment by akurtakov** [comment]
It's quite possible for sure. There are just so many other things that this one stay as it. Last time @LucasBullen worked on making it support JUnit 5 which unfortunately I never managed to finish up.

**Comment by akurtakov** [comment]
@HannesWell I'm closing this one as  test framework can not work without suite to drive it.

**Comment by HannesWell** [comment]
> @HannesWell I'm closing this one as test framework can not work without suite to drive it.

Yes.
I started to work on an enhancement of the `EclipseTestRunner`, but didn't have the time yet to try it out or test if it really works. But I have the intention to come back to that topic later. If that works one day, I will reopen this.

**Pull request closed without merging**